core/src/marker/markerManager.cpp

```diff
@@ -297,6 +298,27 @@ bool MarkerManager::buildStyling(Marker& marker) {
     try {
         // Update the draw rule for the marker.
         YAML::Node node = YAML::Load(marker.stylingString());
+
+        // Check if marker styling defines a layer in the yaml scene
+        if (Node layer = node["layer"]) {
```

Comment:
There really isn't a need to use YAML when specifying a layer. I would suggest creating a separate function along the lines of:
```
bool markerSetLayer(MarkerID _marker, const char* layerName)
```
Then there's no need to read or write the layer name in YAML syntax.